server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/crypto/SecurityKeyLoader.java

```diff
@@ -54,15 +55,25 @@
     public AsymmetricKeys load() throws Exception {
         Preconditions.checkState(jmapDraftConfiguration.isEnabled(), "JMAP is not enabled");
 
-        KeyStore keystore = KeyStore.getInstance(JKS);
+        if (jmapDraftConfiguration.getKeystore().isPresent()) {
+            return loadFromKeystore();
+        }
+        return loadFromPEM();
+    }
+
+    private AsymmetricKeys loadFromKeystore() throws Exception {
+        Preconditions.checkState(jmapDraftConfiguration.getKeystore().isPresent());
+        Preconditions.checkState(jmapDraftConfiguration.getSecret().isPresent());
+
+        KeyStore keystore = KeyStore.getInstance(jmapDraftConfiguration.getKeystoreType());
         char[] secret;
-        try (InputStream fis = fileSystem.getResource(jmapDraftConfiguration.getKeystore())) {
-            secret = jmapDraftConfiguration.getSecret().toCharArray();
+        try (InputStream fis = fileSystem.getResource(jmapDraftConfiguration.getKeystore().get())) {
+            secret = jmapDraftConfiguration.getSecret().get().toCharArray();
             keystore.load(fis, secret);
```

Comment:
By the way, I am not sure if it is worth but you might be able to even refactor line number 68 till 72 into a single line with the KeyStoreUtils:
```
KeyStore keyStore = KeyStoreUtils.loadKeyStore(fileSystem.getResource(jmapDraftConfiguration.getKeystore()), jmapDraftConfiguration.getSecret().get().toCharArray())
```
It will also take care of opening/closing the resource.

Reply:
Mmm propagating a dependency to gain one line...

Reply:
Yes, that was the reason why I wasn't sure to propose it

Reply:
The dependency was already there so I applied your suggestion.

Thanks!